src/main/java/uk/gov/hmcts/reform/cwrdapi/service/ExcelAdaptorServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+                headerToCellValueMap.put(mappingField.columnName(), field);
+            } else { // make triple of child domain object
+                List<Field> customObjectFields = new ArrayList<>();
+                Class childDomainObjectClassType = mappingField.clazz();
+                for (Field childField: childDomainObjectClassType.getDeclaredFields()) {
+                    customObjectFields.add(childField);
+                }
+                // make triple of child domain object class name, parent field, respective list of domain object fields
+                customObjects.add(Triple.of(childDomainObjectClassType.getCanonicalName(), field, customObjectFields));
+            }
+        }
+        return customObjects;
+    }
+
+    private void collectHeaderList(List<String> headers, Sheet sheet) {
+        Row row = sheet.getRow(0);
+        Iterator<Cell> headerIterator = row.cellIterator();
+        while (headerIterator.hasNext()) {
+            Cell cell = headerIterator.next();
+            headers.add(cell.getStringCellValue().trim());
+        }
+    }
+
+    private void setFieldValue(Field field, Object bean, Object value) {
+        if (nonNull(field)) {
+            try {
+                field.setAccessible(TRUE);
+                field.set(bean, value);
+            } catch (IllegalAccessException e) {
+                throwFileParsingException();
+            }
+        }
+    }
+
+    private Object getInstanceOf(String className)  {
+        Object objectInstance = null;
+        try {
+            objectInstance = Class.forName(className).getDeclaredConstructor().newInstance();
+        } catch (Exception e) {
+            throwFileParsingException();
+        }
+        return objectInstance;
+    }
+
+    private void setParentFields(Object cellValue, Object bean, String header, Map<String, Field> fieldHashMap,
+                           Map<String, Object> childHeaderValues) {
+        Field field = fieldHashMap.get(header);
+        if (nonNull(field)) {
+            setFieldValue(field, bean, cellValue);
+        } else {
+            childHeaderValues.put(header, cellValue);
+        }
+    }
+
+    private void setIsPrimaryField(Object childDomainObject, MappingField mappingField, String domainObjectColumnName) {
+        if (nonNull(mappingField.isPrimary()) && domainObjectColumnName.equals(mappingField.isPrimary())) {
+            try {
+                Field primaryField = childDomainObject.getClass().getDeclaredField(IS_PRIMARY_FIELD);
+                setFieldValue(primaryField, childDomainObject, true);
+            } catch (NoSuchFieldException e) {
+                throwFileParsingException();
+            }
+        }
+    }
+
+    private Object getCellValue(Cell cell) {
+        switch (cell.getCellType()) {
+            case STRING:
+                return cell.getStringCellValue();
+            case NUMERIC:
+                return Double.valueOf(cell.getNumericCellValue()).intValue();
```

Comment:
I may wrong but  dint understand the logic why we are converting the numeric value to Double here?.

Reply:
Excel auto coverts the numbers in cell to into Integer ,so we have to explicitly convert